src/test_network.py

```diff
@@ -0,0 +1,25 @@
+import unittest
+from src.network import Network
+
+
+class TestNetwork(unittest.TestCase):
+    net = Network(input_dim=3, units_per_layer=[6, 2], acts=['relu', 'relu'])
+    net.compile('sgd', 'squared', lrn_rate=0.01)
+
+    def test_ExceptionForward(self, net=net):
+        # make sure Exception is raised when there is a mismatch between input_dim and inp
+        self.assertRaises(Exception, net.forward, inp=(1, 1))
+        # make sure Exception is raised when there is a mismatch between targets and outputs
+        self.assertRaises(Exception, net.fit, inp=(1, 1, 1), target=(1, 0, 1))
+
+    def test_ValueError(self):
+        # make sure a ValueError is raised when input_dim or units_per_layer are not positive
+        self.assertRaises(ValueError, Network, input_dim=-2, units_per_layer=[6, 2], acts=['relu', 'relu'])
+        self.assertRaises(ValueError, Network, input_dim=2, units_per_layer=[-4, 0], acts=['relu', 'relu'])
+
+        #TODO: test ValueError if units in one layer have different act functions
+        # This assignment was done with a private method I cannot access for testing
```

Comment:
Which private method?

Reply:
def __init__(self, units):
        """
        Constructor
        :param units: list on layer's units ('Unit' objects)
        """
        units_acts = [u.act.name for u in units]
        for act in units_acts[1:]:
            if act != units_acts[0]:
                raise ValueError("All units in a layer must have the same activation function")
        self.__units = units
        self.__act = self.__units[0].act
        self.__outputs = []

How can I access the parameters of the constructor? Any suggestions?